src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
Yet another suggestion/thought struck me; how about we define the abstract `HyperReal` type such that Real <: HyperReal <: Number? This will have dual benefit; it will allow adhering to the math formality (i.e. allowing Julia to consistently address the matter in its core), while at the same retaining coding simplicity. Then Distributions and any new packages would use HyperReal instead of Real (and GradualNumber could be <: HyperReal). This avoids redefining the Normal as a parametric type by instead doing only this minor change:

```
immutable Normal <: ContinuousUnivariateDistribution
    μ::HyperReal
    σ::HyperReal

    Normal(μ::HyperReal, σ::HyperReal) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
    Normal(μ::HyperReal) = Normal(μ, 1.0)
    Normal() = Normal(0.0, 1.0)
end
```